src/components/Gallery/Gallery.js

```diff
@@ -0,0 +1,345 @@
+    const total = this.props.items.length
+    const srcs = FigureImage.utils.getResizedSrcs(src, window.innerWidth)
+    return (
+      <div {...styles.wrapper}>
+        <Swipeable
+          onSwipedDown={this.handleSwipeDown}
+          onSwipedLeft={this.handleSwipeLeft}
+          onSwipedRight={this.handleSwipeRight}
+          delta={10}
+          preventDefaultTouchmoveEvent={true}
+          stopPropagation={true}
+        >
+          <NavOverlay
+            handleClickLeft={this.handleClickLeft}
+            handleClickRight={this.handleClickRight}
+            handleClick={this.toggleFocus}
+            onClose={onClose}
+          />
+          <div {...styles.gallery}>
+            <div {...styles.header} style={{ opacity: focus ? 0 : 1 }}>
+              <div {...styles.counter}>
+                <span>{index+1}/{total}</span>
+              </div>
+              <div {...styles.close} onClick={onClose}>
+                <Close size={24} />
+              </div>
+            </div>
+            <div {...styles.body}>
+              <div {...merge(
+                styles.mediaItem,
+                exitLeft && styles.exitLeft,
+                exitRight && styles.exitRight,
+                closing && styles.closing)}
+              >
+                <Spinner />
+                <img key={src} alt={alt} {...styles.mediaItemImage} {...srcs} />
+              </div>
+            </div>
+            <div {...styles.caption} style={{ opacity: focus ? 0 : 1 }}>
+              <FigureCaption>
+                { `${caption} ` }
+                { credit && <FigureByline>{ credit }</FigureByline> }
+              </FigureCaption>
+            </div>
+          </div>
+        </Swipeable>
+      </div>
+    )
+  }
+}
+
+Gallery.propTypes = {
+  items: PropTypes.arrayOf(PropTypes.shape({
+    src: PropTypes.string.isRequired,
+    alt: PropTypes.string.isRequired,
+    caption: PropTypes.string,
+    credit: PropTypes.string
+  })),
+  startItem: PropTypes.string,
+  onClose: PropTypes.func.isRequired
+}
+
+Gallery.defaultProps = {
+  items: [],
+}
+
+class NavOverlay extends React.Component {
+  constructor(props) {
+    super(props)
+    this.ref = null
+    this.setRef = ref => this.ref = ref
+    const { handleClickLeft, handleClickRight, onClose } = props
+    this.handleKeyUp = (event) => {
+      switch (event.keyCode) {
+        case 37:
+          handleClickLeft()
+          break;
+        case 39:
+          handleClickRight()
+          break;
+        case 27:
+          onClose()
+          break;
+        default:
+          break;
+      }
+    }
+  }
+  componentDidMount() {
+    this.ref && this.ref.focus()
+  }
+  render() {
+    const { handleClickLeft, handleClickRight, onClose, handleClick } = this.props
+    return (
+      <div
+        ref={this.setRef}
+        {...styles.nav}
+        onKeyDown={e => this.handleKeyUp(e, handleClickLeft, handleClickRight, onClose)}
```

Comment:
I think you don't need to pass `handleClickLeft, handleClickRight, onClose` to `handleKeyUp` here. They're not used from the arguments list there, but from props.